src/main/java/duke/command/Command.java

```diff
@@ -0,0 +1,43 @@
+package duke.command;
+
+import duke.exception.DukeException;
+import duke.storage.Storage;
+import duke.task.TaskList;
+import duke.ui.Ui;
+
+/**
+ * An abstract class that represents a command.
+ */
+public abstract class Command {
+
+    /**
+     * To indicate if the command is an exit command.
+     */
+    protected boolean isExit;
+
+    /**
+     * Abstract method to execute the respective command.
+     *
+     * @param tasks   The user's current TaskList
+     * @param ui      The ui currently being used by the user
+     * @param storage The storage object being used by the user
+     * @throws DukeException An error trying to carry out the command
+     */
+    public abstract void execute(TaskList tasks, Ui ui, Storage storage) throws DukeException;
```

Comment:
It would be nice if Command objects did not access Storage objects directly, but do so through TaskList objects instead. Reason being your implementation does not make use of Storage in the execute method.